src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,423 @@
+            handleInputChange={this.handleInputChange}
+            className="fieldWidthSmaller"
+          />
+          {/* client_ca_crn */}
+          {this.state.method === "certificate" && (
+            <IcseTextInput
+              id={this.props.data.name + "-vpn-server-client-ca-crn"}
+              field="client_ca_crn"
+              componentName="client_ca_crn"
+              labelText="Client Secrets Manager Certificate CRN"
+              value={
+                this.state.client_ca_crn === undefined
+                  ? ""
+                  : String(this.state.client_ca_crn)
+              }
+              onChange={this.handleInputChange}
+              invalid={
+                isNullOrEmptyString(this.state.client_ca_crn) ||
+                invalidCRNs(this.state.client_ca_crn).invalid
+              }
+              invalidText={invalidCRNs(this.state.client_ca_crn).invalidText}
+              className="fieldWidth"
+            />
+          )}
+          {/* client_ip_pool */}
+          <IcseTextInput
+            id={this.props.data.name + "-vpn-server-client-ip-pool"}
+            componentName="client_ip_pool"
+            name="client_ip_pool"
+            field="client_ip_pool"
+            tooltip={{
+              content:
+                "VPN client IPv4 address pool, expressed in CIDR format. The request must not overlap with any existing address prefixes in the VPC or any reserved address ranges.",
+              link: "https://cloud.ibm.com/docs/vpc?topic=vpc-vpn-client-to-site-overview",
+              align: "top-left",
+            }}
+            value={this.state.client_ip_pool}
+            placeholder="x.x.x.x/x"
+            labelText="Client CIDR Pool"
+            invalidCallback={() =>
+              this.props.invalidClientIpPoolCallback(this.state, this.props)
+            }
+            invalidText={this.props.invalidClientIpPoolTextCallback(
+              this.state,
+              this.props
+            )}
+            onChange={this.handleInputChange}
+            className="fieldWidthSmaller"
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          {/* port input */}
+          <NumberInput
+            id={this.props.data.name + "-vpn-server-port"}
+            label="Port"
+            allowEmpty={true}
+            value={this.state.port || ""}
+            step={1}
+            onChange={this.handleNumberInputChange}
+            name="port"
+            hideSteppers={true}
+            min={1}
+            max={65535}
+            invalid={isRangeInvalid(this.state.port, 1, 65535)}
+            invalidText="Must be a whole number between 1 and 65535."
+            className="fieldWidthSmaller leftTextAlign"
+          />
+          {/* protocol */}
+          <IcseSelect
+            formName={this.props.data.name + "-vpn-server-protocol"}
+            groups={["TCP", "UDP"]}
+            value={this.state.protocol.toLowerCase()}
```

Comment:
this dropdown also doesn't work right now because the value that you're setting doesn't match either of the groups - this should be fixed when you implement the above